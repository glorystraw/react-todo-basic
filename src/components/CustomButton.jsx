import React from 'react';
import PropTypes from 'prop-types';

function CustomButton(props) {

    const { onClick, className, icon, children } = props;
    return (
        <button className={className} onClick={onClick} >
            {icon ?
                <i className="material-icons">{icon}</i>
                :
                children
            }
        </button>
    );
}

CustomButton.propTypes = {
    className: PropTypes.string,
    icon: PropTypes.string,
    onClick: PropTypes.func,
    children: PropTypes.node
}

export default CustomButton;