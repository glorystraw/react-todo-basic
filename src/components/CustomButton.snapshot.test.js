import React from 'react';
import renderer from 'react-test-renderer';
import CustomButton from './CustomButton';

test('Snapshot testing enabled Button component', () => {
  const tree = renderer.create(<CustomButton className={'edit icon'} onClick={() => {}}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
