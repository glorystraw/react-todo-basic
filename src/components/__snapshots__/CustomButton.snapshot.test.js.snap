// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing enabled Button component 1`] = `
<button
  className="edit icon"
  onClick={[Function]}
/>
`;
